packages/swapper/src/api.ts

```diff
@@ -33,19 +33,91 @@ export type SupportedSellAssetsInput = {
   assetIds: CAIP19[]
 }
 
+export type CommonTradeInput = {
+  sellAsset: Asset
+  buyAsset: Asset
+  sellAmount?: string
+  buyAmount?: string
+  sendMax: boolean
+}
+export type GetTradeQuoteInput = CommonTradeInput
+
+export type BuildTradeInput = CommonTradeInput & {
+  sellAssetAccountId: string
+  buyAssetAccountId: string
+  slippage?: string
+  sendMax: boolean
+  wallet: HDWallet
+}
+
+export type TradeQuote<C extends ChainTypes> = {
```

Comment:
We tried to make this use a union from common types but it caused strange errors with the generics when using in web.

Happy to look at it further with a typescript expert or merge as-is

Reply:
I think when it comes to types that are specific to a package, we want to keep those types in that package, as opposed to using the common types package. So I think you're fine here.